examples/glue_benchmark/glue.py

```diff
@@ -112,200 +69,127 @@ def split_features(x):
-    for idx, pred in zip(idx_order, predictions):
-        if labelname:
-            pred_value = labelname[int(pred)]
-        else:
-            pred_value = pred
-            if FLAGS.task_name == "stsb":
-                pred_value = min(pred_value, 5)
-                pred_value = max(pred_value, 0)
-                pred_value = f"{pred_value:.3f}"
-        contents[idx] = pred_value
-
-    with tf.io.gfile.GFile(filename, "w") as f:
-        # GLUE requires a format of index + tab + prediction.
-        writer = csv.writer(f, delimiter="\t")
-        # Write the required headline for GLUE.
-        writer.writerow(["index", "prediction"])
-
-        for idx, value in enumerate(contents):
-            writer.writerow([idx, value])
-
-
-def connect_to_tpu(tpu_name):
-    resolver = tf.distribute.cluster_resolver.TPUClusterResolver.connect(
-        tpu=tpu_name
+def main(_):
+    warnings.warn("DISCLAIMER:This script only supports GLUE/mrpc (for now).")
+    # checking task version (erroring out other testes except "mrpc")
+    if FLAGS.task != "mrpc":
+        raise ValueError("task - mrpc is only supported currently.")
+
+    print(tf.__version__)
+    print("GPU available : ", tf.test.is_gpu_available())
+
+    print("=" * 120)
+    print(
+        f"MODEL : {FLAGS.model} | PRESET : {FLAGS.preset} | DATASET : glue/mrpc | batch_size : {FLAGS.batch_size} | epochs : {FLAGS.epochs}"
     )
-    return tf.distribute.TPUStrategy(resolver)
+    print("=" * 120)
 
+    # Load the model
+    model, preprocessor = load_model_and_preprocessor(
+        model=FLAGS.model, preset=FLAGS.preset, num_classes=2
+    )
+    # Add loss and optimizer
+    loss = keras.losses.SparseCategoricalCrossentropy(from_logits=True)
+    metrics = [keras.metrics.SparseCategoricalAccuracy()]
+
+    # Load datasets
+    train_ds, test_ds, validation_ds = load_data()
+    train_ds = preprocess_data(dataset=train_ds, preprocessor=preprocessor)
+    validation_ds = preprocess_data(
+        dataset=validation_ds, preprocessor=preprocessor
+    )
+    print("GLUE/MRPC Dataset Loaded!")
 
-def main(_):
-    if FLAGS.tpu_name:
-        strategy = connect_to_tpu(FLAGS.tpu_name)
-        policy = keras.mixed_precision.Policy("mixed_bfloat16")
-    else:
-        # Use default strategy if not using TPU.
-        strategy = tf.distribute.get_strategy()
-        policy = keras.mixed_precision.Policy("mixed_float16")
-    keras.mixed_precision.set_global_policy(policy)
-
-    train_ds, test_ds, val_ds, idx_order = load_data(FLAGS.task_name)
-    # ----- Custom code block starts -----
-    bert_preprocessor = keras_nlp.models.BertPreprocessor.from_preset(
-        "bert_base_en_uncased"
+    lr = tf.keras.optimizers.schedules.PolynomialDecay(
+        FLAGS.learning_rate,
+        decay_steps=train_ds.cardinality() * FLAGS.epochs,
+        end_learning_rate=0.0,
+    )
+    optimizer = tf.keras.optimizers.experimental.AdamW(
+        lr, weight_decay=0.01, global_clipnorm=1.0
```

Comment:
I will be cautious on the `global_clipnorm`. The glue script i wrote is a template, so I set the value as a reference, we can delete it from this file since it's more of an automatic script.